lib/main.js

```diff
@@ -50,6 +50,80 @@ const SEC_PKCS7EncoderOutputCallback = ctypes.FunctionType(ctypes.default_abi,
                                                             ctypes.char.ptr,
                                                             ctypes.int]);
 
+const PK11SlotInfo = ctypes.voidptr_t;
+// These structure definitions are needed,
+// because there is no function in NSS to get slot from certificate.
```

Comment:
I have tried this but havd problems traversing PK11SlotList from js. Js-ctypes is new for me and I probably made mistake somwere. I will try it again.